File: testing/runtime-errors/visual-basic/runtime_error_1.vb
' Raspberry Pi Foundation
' Developed as part of Ada Computer Science
' Usage licensed under CC BY-NC-SA 4.0
' Note: This file is designed to be copied out and compiled on your machine.
' To run this file you need to:
' 1. Copy the contents
' 2. Paste them into the Visual Basic IDE of your choice (Visual Studio, for example)
' 3. Compile the program
' 4. Run the program

Module Program

    ' The Main subroutine is the default entry point for a VB program
    Sub Main()
        ' Demonstrates a runtime error when entering a string value
        Console.WriteLine("Please enter a number:")
        Dim numberString As String = Console.ReadLine()
        Dim number As Integer = Int32.Parse(numberString)
        Console.WriteLine(number)
    End Sub

End Module

File: testing/runtime-errors/python/runtime-error1.py
# Raspberry Pi Foundation
# Developed to be used alongside Isaac Computer Science,
# part of the National Centre for Computing Education
# Usage licensed under CC BY-SA 4


def main():
    # Demonstrates a runtime error when entering a string value
    number = input("Please enter a number: ")
    number = int(number)
    print(number)


# This code will run if this file is executed directly
# (i.e. not called by another program)
if __name__ == "__main__":
    main()
